src/test/java/org/threeten/extra/scale/TestMispInstant.java

```diff
@@ -0,0 +1,741 @@
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try ( ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(test);
+        }
+        try ( ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(baos.toByteArray()))) {
+            assertEquals(test, ois.readObject());
+        }
+    }
+
+    @Test
+    public void factory_ofMispSecondslong_long() {
+        for (long i = -2; i <= 2; i++) {
+            for (int j = 0; j < 10; j++) {
+                MispInstant t = MispInstant.ofMispSeconds(i, j);
+                assertEquals(i, t.getMispSeconds());
+                assertEquals(j, t.getNano());
+            }
+            for (int j = -10; j < 0; j++) {
+                MispInstant t = MispInstant.ofMispSeconds(i, j);
+                assertEquals(i - 1, t.getMispSeconds());
+                assertEquals(j + 1000000000, t.getNano());
+            }
+            for (int j = 999999990; j < 1000000000; j++) {
+                MispInstant t = MispInstant.ofMispSeconds(i, j);
+                assertEquals(i, t.getMispSeconds());
+                assertEquals(j, t.getNano());
+            }
+        }
+    }
+
+    @Test
+    public void factory_ofMispSeconds_long_long_nanosNegativeAdjusted() {
+        MispInstant test = MispInstant.ofMispSeconds(2L, -1);
+        assertEquals(1, test.getMispSeconds());
+        assertEquals(999999999, test.getNano());
+    }
+
+    @Test
+    public void factory_ofMispSeconds_long_long_tooBig() {
+        assertThrows(ArithmeticException.class, () -> MispInstant.ofMispSeconds(Long.MAX_VALUE, 1000000000));
+    }
+
+    @Test
+    public void factory_of_Instant_null() {
+        assertThrows(NullPointerException.class, () -> MispInstant.of((Instant) null));
+    }
+
+    @Test
+    public void factory_of_UtcInstant_null() {
+        assertThrows(NullPointerException.class, () -> MispInstant.of((UtcInstant) null));
+    }
+
+    @Test
+    public void test_fromTaiInstant() {
+        TaiInstant tai = TaiInstant.parse("378691208.000082000s(TAI)");
+        MispInstant misp = MispInstant.of(tai);
+        assertEquals(0, misp.getMispSeconds());
+        assertEquals(0, misp.getNano());
+    }
+
+    @Test
+    public void test_toTaiInstant() {
+        MispInstant misp = MispInstant.ofMispSeconds(0, 0);
+        TaiInstant tai = misp.toTaiInstant();
+        assertEquals(378691208, tai.getTaiSeconds());
+        assertEquals(82000, tai.getNano());
+    }
+
+    @Test
+    public void test_fromInstant() {
+        Instant instant = Instant.parse("2022-03-05T00:00:08.000082Z");
+        MispInstant misp = MispInstant.of(instant);
+        // 37 seconds for TAI to UTC offset, and 8.000082 for MISP to TAI offset
+        // 59643L is the MJD for 2022-03-05
+        // 40587L is the MJD for 1970-01-01
+        assertEquals((59643L - 40587L) * 24 * 60 * 60 + 37, misp.getMispSeconds());
+        assertEquals(0, misp.getNano());
```

Comment:
I'm not seeing the 8.000082 anywhere?

Reply:
The 8.000082 is in the `instant` that we are converting. I can rework the example (or the comment) if it isn't clear.